app/services/github_handler.rb

```diff
@@ -0,0 +1,84 @@
+class GithubHandler
+  RETURNS = [
+    SUCCESS = :success,
+    FORBIDDEN = :forbidden,
+    BAD_REQUEST = :bad_request
+  ].freeze
+
+  attr_reader :event, :payload, :raw_payload, :signature
+
+  def initialize(event, payload, signature)
+    @event = event
+    @payload = JSON.parse(payload)
+    @raw_payload = payload
+    @signature = signature
+  end
+
+  def handle
+    return FORBIDDEN unless webhook_verified?
+
+    case event
+    when 'pull_request'
+      case payload['action']
+      when 'review_requested'
+        handle_review_request
+        # when "review_request_removed"
+        # TODO: Call the method to handle a review request removal
+      else
+        return BAD_REQUEST
+      end
+    else
+      return BAD_REQUEST
+    end
+    SUCCESS
+  rescue ActiveRecord::RecordInvalid
+    BAD_REQUEST
+  end
+
+  def handle_review_request
+    owner = create_or_find_user(payload['pull_request']['user'])
+    pr = create_or_find_pr(payload['pull_request'])
+    # Even if you select multiple reviewers at once,
+    # the webhook sends a post for every person selected
+    # we can assume it is going to be a single user obj "requested_reviewer"
+    reviewer = create_or_find_user(payload['requested_reviewer'])
+    pr.review_requests.create!(data: payload, owner: owner, reviewer: reviewer)
+  end
+
+  def webhook_verified?
+    digest = OpenSSL::HMAC.hexdigest('SHA1', ENV['GITHUB_ANALYZER_WEBHOOK_SECRET'], raw_payload)
+    ActiveSupport::SecurityUtils.secure_compare("sha1=#{digest}", signature)
+  end
+
+  def create_or_find_user(json)
+    user = User.find_by(github_id: json['id'])
+    if user.nil?
+      user = User.create!(
+        node_id: json['node_id'],
+        login: json['login'],
+        github_id: json['id']
+      )
+    end
+    user
+  end
+
+  def create_or_find_pr(json)
+    pr = PullRequest.find_by(json['github_id'])
+    if pr.nil?
```

Comment:
Also I think you can just do something like `PullRequest.find_by(...) || PullRequest.create!(....)`